Tests/SwiftSMTPTests/TestHeaderFolding.swift

```diff
@@ -0,0 +1,136 @@
+
+    var previousLogger: Logger?
+    var currentLogger: TestLogger?
+
+    override func setUpWithError() throws {
+        previousLogger = Log.logger
+        currentLogger = TestLogger()
+        Log.logger = currentLogger
+    }
+
+    override func tearDownWithError() throws {
+        Log.logger = previousLogger
+        currentLogger = nil
+    }
+
+    // Here we test two things: 1) a short line will not be folded,
+    // and 2) a line containing multiple consecutive whitespace characters
+    // will not be changed.
+    func testShortHeaderUnchanged() throws {
+        let allHeaders = mailMessage.headersString.split(separator: "\r\n")
+        let unmodified = "X-SHORT: \(spaciousSubject)"
+        for header in allHeaders where header.hasPrefix("X-SHORT") {
+            XCTAssertEqual(String(header), unmodified)
+        }
+    }
+
+    // Whatever else is true of the folded header, the folding process should
+    // not ever produce a line which consists solely of whitespace.
+    func testNoWSOnlyLines() throws {
+        let allHeaders = mailMessage.headersString.split(separator: "\r\n")
+        for header in allHeaders {
+            XCTAssert(!header.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty)
+        }
+    }
+
+    // This test looks at folding a long name plus email address, which won't
+    // contain a comma. The fold should happen at the space between the name
+    // and the email.
+    func testAbsurdEmail() throws {
+        let allHeaders = mailMessage.headersString.split(separator: "\r\n")
+        var foundCC = false
+        for header in allHeaders {
+            if foundCC {
+                // we are now at the first line *after* the CC: line
+                XCTAssertEqual(header, " <unfortunate_joe_1234567890_123456789_123456783_123456784_123456785_123456786@dumbster.local>")
+                break
+            } else  if header.hasPrefix("CC:") {
+                foundCC = true
+                XCTAssertEqual(header, "CC: \("Unfortunate Joe".mimeEncoded!) ")
+                continue
+            }
```

Comment:
I think this misses the potential case where there is no header after a "CC".

Reply:
The test message has, in addition to FROM, TO, CC, and SUBJECT, two additional headers, X-OBNOXIOUS and X-SHORT. Are you suggesting the test be rewritten to account for the case where the CC header appears last in the list? I guess that's reasonable.

Reply:
Yeah, I'm not sure it'd require a rewrite.  It might just need an extra flag to ensure that the foundCC block got executed.  It's not "pretty" but would be sufficient.